### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by Vinnl**
# New feature description

This moves us away from Whitesource, which does not work reliably
and takes a long time to run, and replaces it by `npm audit`.
Additionally, it runs the unit tests, licence checks, a test build and Prettier verification in GitHub Actions.

After this is merged, I'll modify the GitHub settings to require these checks to succeed.

The next step will be moving over the graphical tests, then
removing the Travis build.

# Checklist

- [ ] All acceptance criteria are met. (Other PR will finish them.)
- [x] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      uses: actions/cache@v1.1.2
+      env:
+        cache-name: cache-node-modules
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+    - run: npm ci
+    - run: npm run build
+    - run: npm test
+    - run: npx prettier --check "src/**"
```
Comment (Vinnl) [inline .github/workflows/ci.yml:31]: @NSeydoux I wasn't sure about this line - is all the relevant, Prettier'd source code in `src`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - run: npm audit --audit-level=moderate
```
Comment (Vinnl) [inline .github/workflows/ci.yml:32]: Should we run this in the `examples` dirs as well, you think?

**Comment by Vinnl** [comment]
The ESLint setup was wrong and therefore did not actually run (but this was only detectable in Windows, on which the tests now also run). See aebccb9d78473de751593a9cb2a9c713c48d2f22 for a description of what was wrong, and the fix.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (NSeydoux) [reply .github/workflows/ci.yml:31]: Some of it is under ``__tests__` as well, even if I'd like to move the test code alongside the tested files at some point.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (NSeydoux) [reply .github/workflows/ci.yml:32]: Given the examples aren't shipped with the code, I'm not sure it is necessary, but if it's easy to do and does not add noise that seems reasonable. I think I'd try it to see how noisy it turns out to be, only to disable it if the cost outwheights the benefits.

File: src/login/oidc/redirectHandler/InactionRedirectHandler.ts [hunk 3]
```diff
@@ -27,13 +27,17 @@ import { ISessionInfoManager } from "../../../sessionInfo/SessionInfoManager";
 @injectable()
 export default class InactionRedirectHandler implements IRedirectHandler {
   constructor(
+    // FIXME: I encountered a number of ESLint issues that had been missed due to an invalid setup,
+    //        and I'm unsure whether TSyringe depends on a specific variable name.
```
Comment (NSeydoux) [inline src/login/oidc/redirectHandler/InactionRedirectHandler.ts:31]: `TSyringe` should only depend on the `@inject("sessionInfoManager")` notation, but when splitting up the codebase I'll clean up these unsused dependencies.

File: .eslintrc.json [hunk 4]
```diff
@@ -14,6 +14,7 @@
     "prettier/prettier": "error",
     "@typescript-eslint/interface-name-prefix": ["error", "always"],
     "import/no-unresolved": 0,
-    "license-header/header": ["error", "./resources/license-header.js"]
+    "license-header/header": ["error", "./resources/license-header.js"],
+    "@typescript-eslint/no-unused-vars": ["warn", { "argsIgnorePattern": "^_" }]
```
Comment (NSeydoux) [inline .eslintrc.json:18]: Good to know !

File: src/login/oidc/redirectHandler/InactionRedirectHandler.ts (hunk 3 above)
Comment (Vinnl) [reply src/login/oidc/redirectHandler/InactionRedirectHandler.ts:31]: I don't know why I can't respond to your other comments, but:

- Prettier on tests: 0fb275af6f2d8a8b381d2889d5212767143e9324
- Audit of examples (barely any noise): b6a94d0a437e67359a1e077defb61a4c92cf9b55
- Unused injected dependency: 76283dcacf82bd7e5722580a2d27aece9b7d3163

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Vinnl) [reply .github/workflows/ci.yml]: 0fb275af6f2d8a8b381d2889d5212767143e9324

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Vinnl) [reply .github/workflows/ci.yml:32]: Barely any noise: b6a94d0a437e67359a1e077defb61a4c92cf9b55

**Comment by Vinnl** [comment]
@NSeydoux I just found out that there were no graphical tests run in Travis, which means that with this PR, the entire Travis setup has been replaced. Thus, I removed the Travis config as well, and re-requested a review to make sure I did not forget anything :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Vinnl) [reply .github/workflows/ci.yml:32]: Ah, but apparently it doesn't like the relative dependency. I'll remove it then.

**Pull request merged**